client_connection/client_connection.go

```diff
@@ -0,0 +1,55 @@
+package client_connection
+
+import (
+	"errors"
+	"github.com/mysterium/node/service_discovery/dto"
+)
+
+type State string
+
+const (
+	NOT_CONNECTED = State("NOT_CONNECTED")
+	NEGOTIATING   = State("NEGOTIATING")
+	CONNECTED     = State("CONNECTED")
+)
+
+var (
+	ALREADY_CONNECTED = errors.New("already connected")
+)
+
+type Status struct {
+	State     State
+	SessionId string
+}
+
+type Manager interface {
+	Connect(identity dto.Identity, NodeKey string) error
+	Status() Status
+	Disconnect() error
+	Wait() error
+}
+
+type nullManager struct {
```

Comment:
Wer kinda had pattern to call such dummy stubs: `nullManager` -> `ManagerFake`

Reply:
This component is still in progress and will be eventually renamed in other branch